lib/sandbox/dispatch-event.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Why Do we need this wrapper?
+ * Because when user executes pm.request.stopExecution(), we need to stop the execution of the current request.
+ * But, we don't stop the execution of the script. We just stop sending events to the bridge.
+ *
+ * @param {any} bridge - the bridge object
+ * @param {{ shouldSkipExecution: boolean }} execution - the execution object
+ * @param {string} event - the event name
+ * @param  {...any} args - the arguments to be passed to the event
+ */
+module.exports = function dispatchEvent (bridge, execution, event, ...args) {
+    // if the execution is skipped, do not dispatch the event
+    if (execution && execution.shouldSkipExecution) {
+        return;
+    }
+
+    bridge.dispatch(event, ...args);
```

Comment:
What're your thoughts on overriding the `dispatch` method itself? Would solve the problem where consumers can accidentally call `.dispatch` directly on the bridge.

```javascript
const originalDispatch = bridge.dispatch;

bridge.dispatch = function (execution, ...args) {
  if (!isExecution(execution)) {
    throw new Error('must be called with execution');
  }

  if (execution.shouldSkipExecution) {
    return;
  }

  originalDispatch.call(bridge, ...args);
};

```